migrations/subfolderrights.js

```diff
@@ -0,0 +1,56 @@
+/* eslint no-console: 0 */
+/* eslint no-confusing-arrow: 0 */
+const ran = false; // set to true to exclude migration
+const name = 'Fix access rights for teachers on homework submitted files.';
+
+const database = require('../src/utils/database');
+
+const RoleModel = require('../src/services/role/model');
+const { FileModel } = require('../src/services/fileStorage/model.js');
+
+
+const run = async (dry) => {
+	database.connect();
+
+	const errorHandler = (e) => {
+		console.log('Error', e);
+		return undefined;
+	};
+
+	const { _id: studentRoleId } = await RoleModel.findOne({ name: 'student' }).lean().exec();
+
+	const courseFolders = await FileModel
+		.find({
+			isDirectory: true,
+			refOwnerModel: 'course',
+		})
+		.exec()
+		.catch(errorHandler);
+
+	if (!courseFolders.length) {
+		return Promise.resolve();
+	}
+
+	const promises = courseFolders
+		.map((dir) => {
+			const { permissions, _id } = dir;
+			const studentPermissions = permissions.find(perm => perm.refId && perm.refId.equals(studentRoleId));
+
+			if (!studentPermissions.create) {
+				studentPermissions.create = true;
```

Comment:
Imho, it would be nicer if you wouldn't use a side-effect here, but explicitly use an updated permissions set, á la `updatedPermissions = [...withoutStudents(permissions), newStudentPermissions]`

Reply:
Can you elaborate which side-effects you are spotting here?

Reply:
sorry for the confusion. I misread your code 🙊